### PR TITLE
feat: method calculating refund amount

### PR DESCRIPTION
**Pull request by pawel-betdex**
adding method that allows client to determine amount of refund based on order and order's owner market position.

### CODE REVIEW

File: npm-client/src/cancel_order.ts [hunk 1]
```diff
@@ -115,3 +116,66 @@ export async function cancelOrdersForMarket(
...
+ * @param order {Order} order to be canceled
+ * @param marketPosition {MarketPosition} market position of the order's owner
+ * @returns the amount of the refund in raw form; this means it needs to be divided by the mint decimals before it can be dispalyed
+ *
+ * @example
+ *
+ * const marketPk = new PublicKey('7o1PXyYZtBBDFZf9cEhHopn2C9R4G6GaPwFAxaNWM33D')
+ * const cancelledOrders = await cancelOrdersForMarket(program, marketPk)
+ */
+export function cancelOrderRefund(
```
Comment (eoin-betdex) [inline npm-client/src/cancel_order.ts:132]: I'd maybe suggest renaming this function. The other functions that start with `cancel...` actually trigger an on chain instruction, whereas this is just getting/calculating/estimating a value.

**Review by eoin-betdex — COMMENTED** [review]
LGTM but think a different function name would be better.

**Comment by eoin-betdex** [comment]
If we're happy with this one today we can hopefully get it included in the v0.13.0 release branch.

File: npm-client/src/cancel_order.ts (hunk 1 above)
Comment (pawel-betdex) [reply npm-client/src/cancel_order.ts:132]: `calculateOrderCancellationRefund` ?

**Review by eoin-betdex — APPROVED** [review]
LGTM a couple of comments but nothing blocking

File: npm-client/src/cancel_order.ts [hunk 2]
```diff
@@ -115,3 +116,69 @@ export async function cancelOrdersForMarket(
...
+}
+
+function totalExposure(
+  unmatchedExposures: number[],
+  matchedExposures: number[],
+): number {
+  const minLength = Math.min(
+    matchedExposures.length,
+    unmatchedExposures.length,
+  );
```
[the remark this answers is not in the document]
Comment (pawel-betdex) [reply npm-client/src/cancel_order.ts:176]: They are.

**Review by eoin-betdex — APPROVED** [review]
LGTM

**Review by obi-ewan — APPROVED** [review]

**Pull request merged**